design/one-pager-easy-stacks.md

```diff
@@ -0,0 +1,294 @@
+```
+
+Now, I want an environment where all my database instances and kubernetes clusters
+are connected to each other in a private VPC, which what this specific configuration
+stack does. Create the following:
+
+```yaml
+# exact apiVersion is TBD.
+apiVersion: aws.configurationstacks.crossplane.io/v1alpha1
+kind: EasyAWS
+metadata:
+  name: project-future
+  namespace: dev
+spec:
+  providerRef:
+    name: aws-provider
+```
+
+Then I wait for status to become ready. After it's done, all resources are deployed.
+```yaml
+# exact apiVersion is TBD.
+apiVersion: aws.configurationstacks.crossplane.io/v1alpha1
+kind: EasyAWS
+metadata:
+  name: project-future
+  namespace: dev
+spec:
+  providerRef:
+    name: aws-provider
+status:
+  state: ready
+```
+
+An example deployed resource would be:
+```yaml
+apiVersion: database.aws.crossplane.io/v1beta1
+kind: RDSInstanceClass
+metadata:
+  name: dev-project-future-mysql
+  labels:
+    "aws.configurationstacks.crossplane.io/name": project-future
+    "aws.configurationstacks.crossplane.io/namespace": dev
+    "aws.configurationstacks.crossplane.io/uid": 1ca16960-973b-4d58-a6fa-5696638ec631
+specTemplate:
+  writeConnectionSecretsToNamespace: crossplane-system
+  forProvider:
+    dbInstanceClass: db.t2.small
+    masterUsername: root
+    vpcSecurityGroupIDRefs:
+      - name: dev-project-future-rds-security-group
+    dbSubnetGroupNameRef:
+      name: dev-project-future-dbsubnetgroup
+    allocatedStorage: 20
+    engine: mysql
+    skipFinalSnapshotBeforeDeletion: true
+  providerRef:
+    name: aws-provider
+  reclaimPolicy: Delete
+```
+
+Other people in different namespaces can create their own instances of `EasyAWS`
+custom resource and have their own similar environment with different names.
+
+## Technical Implementation
+
+### Metadata of The Stack
+
+[Dependency list] has to be filled with each kind of the resource that the stack
+deploys, i.e. f the stack only deploys `RDSInstanceClass` resource, it should
+not require all AWS stack to be present.
+
+[UI annotations] should be present to make it easy for frontend software to process
+the stack.
+
+### Folder Structure
+
+All directories containing base YAML files should be in a top-level directory
+of the repository named `resources` so that it's easier for others to fork & change.
+
+Controller will scan this folder recursively to get all YAML files. So, the folder
+structure inside that `resources` directory won't matter to the controller but
+it's good to separate the folders cleanly so that it's easier to fork & change.
+
+There are mainly 2 types of resources that configuration stacks will usually
+need to deploy; resource classes and managed resources. Under `resources` folder
+we can have 2 separate folders for each of those resource types; `classes` and
+`services`. Under these folders, resources should be in directories recursively
+named after their group and kind. An example folder structure:
```

Comment:
Maybe I am misunderstanding, but it doesn't seem like the example folder structure matches "`directories recursively named after their group and kind`". It seems more like the folder is the group, and the filename is `{{kind}}.yaml`. Am I reading the example correctly?

Reply:
I am not 100% sure about this one actually. Going as deep as having a folder for each `kind` looked too verbose to me. What do you think about having `resources/classes/<groupname>/<resource name>.yaml` ? For resources of different kinds with same names, the author probably would want to make the file name something like `<kind>-<object name>.yaml`

Reply:
I don't think we need to enforce any particular structure. I would provide some examples of what we feel are best practices for logical groupings, but keep the tooling unopinionated and leave it up to the author to decide what works well for them. If it's supported by the tool they're using underneath (in this case, kustomize), then we should support it. The grouping that we're using in the example structure looks pretty reasonable to me.